File: src/main/java/unificado/core/service/repository/NegocioRepository.java
package unificado.core.service.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import unificado.core.service.model.Negocio;

import java.util.Optional;

public interface NegocioRepository extends JpaRepository<Negocio, Integer> {

    Optional<Negocio> findByNegocioId(Integer negocioId);
}

File: src/main/java/unificado/api/rest/kotlin/repository/NegocioRepository.kt
package unificado.api.rest.kotlin.repository

import org.springframework.data.jpa.repository.JpaRepository
import unificado.api.rest.kotlin.model.Negocio
import java.util.Optional

interface NegocioRepository : JpaRepository<Negocio, Int> {

    fun findByNegocioId(negocioId: Int): Optional<Negocio?>?

}